packages/compiler-core/src/parse.ts

```diff
@@ -237,12 +242,8 @@ function parseChildren(
           node.content = node.content.replace(/[\t\r\n\f ]+/g, ' ')
         }
       }
-      // also remove comment nodes in prod by default
-      if (
-        !__DEV__ &&
-        node.type === NodeTypes.COMMENT &&
-        !context.options.comments
-      ) {
+      // comment nodes handling
+      if (node.type === NodeTypes.COMMENT && !context.options.comments) {
```

Comment:
@HcySunYang is it against or in line with Vue coding standarts to use `else if` here for performance reasons?